### PR TITLE
Feature/refactor error message string view

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

Make `Result::errorMessage()` return a `string_view`, instead of a `string`. This avoids unnecessary copies in several places. Also, it is useful for another upcoming change to `Result`.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [X] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [X] No backports required
- [ ] Backports required for: *(Please specify versions and link PRs)*

### Testing & Verification

*(Please pick either of the following options)*

- [X] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [X] This change is already covered by existing tests, such as *scripts/unittest*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/13740/

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/13741/

**Review by maierlars — APPROVED** [review]
I tried to look at all places where used but is it safe to return a `string_view` now instead of a `string`? I'm thinking about ownership. Maybe there are places that use this function but are not shown in the diff.

**Comment by goedderz** [comment]
> I tried to look at all places where used but is it safe to return a `string_view` now instead of a `string`? I'm thinking about ownership. Maybe there are places that use this function but are not shown in the diff.

Yes, it is not safe in general, there may be usages where it isn't. I'll go through other callers once before going out of Draft, and I've started an ASan Jenkins:

http://jenkins01.arangodb.biz:8080/job/arangodb-ANY-linux-asan/534/

**Comment by goedderz** [comment]
I've checked all callers 🥱, and everything seemed obviously safe.

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/13751/
http://jenkins01.arangodb.biz:8080/job/arangodb-ANY-linux-asan/535/

File: lib/Basics/ResultT.h [hunk 1]
```diff
@@ -223,7 +223,8 @@ class ResultT {
   bool fail() const { return _result.fail(); }
   bool is(int code) { return _result.is(code); }
   int errorNumber() const { return _result.errorNumber(); }
-  std::string errorMessage() const { return _result.errorMessage(); }
+  [[nodiscard]] std::string_view errorMessage() const& { return _result.errorMessage(); }
+  [[nodiscard]] std::string errorMessage() && { return std::move(_result).errorMessage(); }
```
Comment (goedderz) [inline lib/Basics/ResultT.h:227]:
```suggestion
  [[nodiscard]] std::string&& errorMessage() && { return std::move(_result).errorMessage(); }
```

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/13753/
http://jenkins01.arangodb.biz:8080/job/arangodb-ANY-linux-asan/536/

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/13756/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-ANY-linux-asan/537/

**Comment by goedderz** [comment]
image

**Comment by goedderz** [comment]
Do not merge before https://github.com/arangodb/arangodb/pull/13440 is merged, and the base branch here is changed to `devel` again.

**Review by maierlars — APPROVED** [review]

**Pull request merged**